test/integration/node-specific/mongo_client.test.ts

```diff
@@ -381,4 +373,150 @@ describe('MongoClient integration', function () {
+        expect(findCommandStarted).to.have.property('commandName', 'find');
+        expect(client).to.have.property('topology').that.is.instanceOf(Topology);
+
+        await client.close();
+      }
+    );
+  });
+
+  it(
+    'should automatically connect upon first operation (find)',
+    { requires: { auth: 'enabled' } },
+    async function () {
+      const client = this.configuration.newClient(this.configuration.url(), {
+        monitorCommands: true
+      });
+
+      const findCommandToBeStarted = once(client, 'commandStarted');
+      await client.db().collection('test').findOne();
+      const [findCommandStarted] = await findCommandToBeStarted;
+
+      expect(findCommandStarted).to.have.property('commandName', 'find');
+      expect(client.options).to.not.have.property(Symbol.for('@@mdb.skipPingOnConnect'));
+      expect(client.s.options).to.not.have.property(Symbol.for('@@mdb.skipPingOnConnect'));
+
+      // Assertion is redundant but it shows that no initial ping is run
+      expect(findCommandStarted.commandName).to.not.equal('ping');
+
+      expect(client).to.have.property('topology').that.is.instanceOf(Topology);
+
+      await client.close();
+    }
+  );
+
+  it(
+    'should automatically connect upon first operation (insertOne)',
+    { requires: { auth: 'enabled' } },
+    async function () {
+      const client = this.configuration.newClient(this.configuration.url(), {
+        monitorCommands: true
+      });
+
+      const insertOneCommandToBeStarted = once(client, 'commandStarted');
+      await client.db().collection('test').insertOne({ a: 1 });
+      const [insertCommandStarted] = await insertOneCommandToBeStarted;
+
+      expect(insertCommandStarted).to.have.property('commandName', 'insert');
+      expect(client.options).to.not.have.property(Symbol.for('@@mdb.skipPingOnConnect'));
+      expect(client.s.options).to.not.have.property(Symbol.for('@@mdb.skipPingOnConnect'));
+
+      // Assertion is redundant but it shows that no initial ping is run
+      expect(insertCommandStarted.commandName).to.not.equal('ping');
+
+      expect(client).to.have.property('topology').that.is.instanceOf(Topology);
+
+      await client.close();
+    }
+  );
+
+  it(
+    'should pass connection errors to the user through the first operation',
+    { requires: { auth: 'enabled' } },
+    async function () {
+      const client = this.configuration.newClient(
+        'mongodb://iLoveJavascript?serverSelectionTimeoutMS=100',
+        { monitorCommands: true }
+      );
+
+      const result = await client
+        .db('test')
+        .collection('test')
+        .findOne()
+        .catch(error => error);
+
+      expect(result).to.be.instanceOf(MongoServerSelectionError);
+      expect(client).to.be.instanceOf(MongoClient);
+      expect(client).to.have.property('topology').that.is.instanceOf(Topology);
+
+      await client.close();
+    }
+  );
+
+  it(
```

Comment:
Should we test that 1) a user can explicitly reconnect and 2) a user can perform an operation after the client is closed, causing the client to implicitly reconnect?